lib/utils.pm

```diff
@@ -447,15 +447,13 @@ sub assert_shutdown_and_restore_system {
     $shutdown_timeout //= 60;
     my $vnc_console = get_required_var('SVIRT_VNC_CONSOLE');
     console($vnc_console)->disable_vnc_stalls;
-    # Ideally, `assert_shutdown` is here instead of `sleep`, but poo#16418
-    # prevent us to be it so. Rather wait couple of seconds more than risk
-    # an 2 hour hang.
-    sleep($shutdown_timeout);
+    assert_shutdown($shutdown_timeout);
     if ($action eq 'reboot') {
         reset_consoles;
         # Set disk as a primary boot device
         console('svirt')->change_domain_element(os => boot => {dev => 'hd'});
         console('svirt')->define_and_start;
+        console('svirt')->start_serial_grab;
```

Comment:
Is this necessary? The `define_and_start` function is already invoking start_serial_grab.

Reply:
Makes sense, thanks.